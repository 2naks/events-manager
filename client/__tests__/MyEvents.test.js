import ConnectedMyEvents, { MyEvents } from '../components/MyEvents.jsx';

describe('MyEvents Component', () => {
  const initialState = {
    myEvents: {
      data: {
        rows: []
      }
    },
    eventReducer: {
      userEvents: {
        data: {
          rows: [],
          page: {}
        }
      }
    },
  };


  const store = mockStore(initialState);

  global.localStorage = {
    getItem: (str) => str
  };

  describe('MyEvents form  should render self', () => {
    it('it should render for dumb component', () => {
      const wrapper = mount(<MyEvents />);
      expect(wrapper).toMatchSnapshot();
    });

    it('it should render component', () => {
      const wrapper = mount(<Provider store={store}><ConnectedMyEvents /></Provider>);
      expect(wrapper.length).toEqual(1);
    });
  });
});
