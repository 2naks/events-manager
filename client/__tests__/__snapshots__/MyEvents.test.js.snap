// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyEvents Component MyEvents form  should render self it should render for dumb component 1`] = `
<MyEvents>
  <div>
    <NavBar
      auth={true}
      page="MyEvents"
    >
      <nav
        className="navbar navbar-toggleable-md navbar-light bg-faded"
      >
        <button
          aria-controls="navbarText"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler navbar-toggler-right"
          data-target="#navbarText"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <a
          className="navbar-brand"
          href="/"
        >
          EVENTS MANAGER
        </a>
        <div
          className="collapse navbar-collapse"
          id="navbarText"
        >
          <ul
            className="navbar-nav ml-auto"
          >
            <li
              className="nav-item active"
            >
              <a
                className="nav-link"
                href="/"
              >
                Home 
                <span
                  className="sr-only"
                >
                  (current)
                </span>
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/centers"
              >
                Centers
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/login"
              >
                Sign In
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/signup"
              >
                Sign Up
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </NavBar>
    <div
      className="container events"
    >
      <div
        className="row event-row"
      />
    </div>
  </div>
</MyEvents>
`;
